lookout/style/format/benchmarks/evaluate_smoke.py

```diff
@@ -1,48 +1,190 @@
+        if action == "delete":
+            res1.append(seq1[i:i_end])
+            empty = EMPTY * (i_end - i)
+            res2.append(empty)
+            if seq3:
+                res3.append(empty)
+    if seq3:
+        return "".join(res1), "".join(res2), "".join(res3)
+    return "".join(res1), "".join(res2)
+
+
+def align3(seq1: Sequence, seq2: Sequence, seq3: Sequence) -> Tuple[Sequence, Sequence, Sequence]:
+    """
+    Align three sequences using levenshtein distance.
+
+    For example:
+    In[1]: align("aabc", "abbcc", "ccdd")
+    Out[1]: ("aab␣c␣␣␣",
+             "␣abbcc␣␣",
+             "␣␣␣␣ccdd")
+
+    It can be suboptimal because heuristic is used because true calculation requires
+    ~ len(seq1) * len(seq2) * len(seq3) time.
+
+    :param seq1: First sequence to align.
+    :param seq2: Second sequence to align.
+    :param seq3: Third sequence to align.
+    :return: Aligned sequences
+    """
+    aseq1, aseq2 = align(seq1, seq2)
+    res3, res1, res2 = align(seq3, aseq1, aseq2)
+    return res1, res2, res3
 
-    return edit_distance(predicted_lines, correct_lines) / \
-        edit_distance(head_lines, correct_lines)
+
+def metrics(init: str, correct: str, model_out: str) -> Tuple[int, int, int, int]:
+    """
+    Calculate quality metrics for aligned sequences.
+
+    Metrics description:
+    1. Amount of characters misdetected by the model as a style mistake. That is nothing needed to
+       be changed but model did.
+    2. Amount of characters undetected by model. That is the character needed to be changed
+       but model did not.
+    3. Amount of characters detected by model as a style mistake but fix was wrong. That is
+       the character needed to be changed and model did but did it wrongly.
+    4. Amount of characters detected by model as a style mistake and fix was correct. That is
+       the character needed to be changed model did it in a correct way :tada:.
+
+    :param init: Initial file with style violations from head revision.
+    :param correct: File with correct style from base revision.
+    :param model_out: Format Analyser model output. File with fixed style.
+
+    :return: Tuple with 4 metric values.
+    """
+    detected_bad_change = 0
+    detected_good_change = 0
+    misdetection = 0
+    undetected = 0
+    for init_c, correct_c, model_out_c in zip(init, correct, model_out):
+        if init_c == correct_c == model_out_c:
+            continue
+        elif init_c == correct_c and init_c != model_out_c:
+            misdetection += 1
+        elif init_c == model_out_c and init_c != correct_c:
+            undetected += 1
+        elif correct_c == model_out_c and init_c != correct_c:
+            detected_good_change += 1
+        else:
+            detected_bad_change += 1
+
+    return misdetection, undetected, detected_bad_change, detected_good_change
+
+
+def _losses(init_file, correct_file, fe, vnodes_y, y_pred, vnodes, url, commit):
```

Comment:
calc_losses? update_losses? There should always be a verb in the function name. BTW, would be cool if there was a docstring here, although not critical.

This is the first protected function which I see, so maybe make it public then?